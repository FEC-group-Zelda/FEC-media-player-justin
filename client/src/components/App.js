import React from 'react';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      media: []
    }
  }

  render() {
    return (
      <div>
        <h1>App Component rendering</h1>
      </div>
    )
  }
}

export default App;