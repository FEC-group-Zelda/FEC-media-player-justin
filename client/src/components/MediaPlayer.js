import React from 'react';
import moment from 'moment';
import Container from '../css-modules/Container.js';
import Image from '../css-modules/Image.js';
import Date_Genre_Style from '../css-modules/Date-Genre.js';
import Band from '../css-modules/BandName.js';
import SongStyle from '../css-modules/SongName.js';

const MediaPlayer = ({ song }) => {
  const now = moment();
  const created = moment(song.release_date);
  const timeElapsed = created.from(now);

  return (
    <Container>
      <Image src={song.album_image} />
      <Date_Genre_Style>
        {timeElapsed}<br />
        {song.music_genre}
      </Date_Genre_Style>
      <audio controls src={song.song_url}></audio>
      <Band>{song.band_name}</Band>
      <SongStyle>{song.song_name}</SongStyle>
    </Container>
  )
}

export default MediaPlayer;