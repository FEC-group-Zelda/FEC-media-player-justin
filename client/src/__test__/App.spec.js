import React from 'react';
import { shallow, mount } from 'enzyme';

import App from '../components/App.js';
import MediaPlayer from '../components/MediaPlayer.js';

describe('App', () => {
  test('should render snapshot', () => {
    const wrapper = mount(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<MediaPlayer />', () => {
  const song = {
    song_name: '10,000 Hours',
    music_genre: '#Pop',
    band_name: 'Dan + Shay',
    album_image: 'https://fec-media-player.s3.us-east-2.amazonaws.com/10%2C000.jpg',
    song_url: 'https://fec-media-player.s3.us-east-2.amazonaws.com/Dan+%2B+Shay%2C+Justin+Bieber+-+10%2C000+Hours+(LYRICS).mp3',
    release_date: '3 October 2019'
  };
  test('should accepts song props', () => {
    const wrapper = mount(<MediaPlayer song={song} />);
    expect(wrapper.props().song).toEqual(song);
  });
});

describe('Play button', () => {
  const mockFn = jest.fn();
  test('play button should be defined', () => {
    expect(PlayButton).toBeDefined();
  });
  test('should render play button correctly', () => {
    const wrapper = shallow(<PlayButton name='play button test' />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should have a play button value', () => {
    const wrapper = shallow(<PlayButton name='play button test' />);
    expect(typeof(wrapper.find('.play').node.props.value)).toBe('string');
    expect(wrapper.find('.play').node.props.value).toEqual('play button test');
  });
  test('should call mock function when play button is clicked', () => {
    const wrapper = shallow(<PlayButton name='play button test' handClick={mockFn} />);
    wrapper.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});

describe('Pause button', () => {
  const mockFn = jest.fn();
  test('pause button should be defined', () => {
    expect(PauseButton).toBeDefined();
  });
  test('should render pause button correctly', () => {
    const wrapper = shallow(<PauseButton name='pause button test' />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should have a pause button value', () => {
    const wrapper = shallow(<PauseButton name='pause button test' />);
    expect(typeof(wrapper.find('.pause').node.props.value)).toBe('string');
    expect(wrapper.find('.pause').node.props.value).toEqual('pause button test');
  });
  test('should call mock function when pause button is clicked', () => {
    const wrapper = shallow(<PauseButton name='pause button test' handClick={mockFn} />);
    wrapper.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});