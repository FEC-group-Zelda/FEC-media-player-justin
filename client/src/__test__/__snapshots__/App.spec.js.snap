// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render snapshot 1`] = `ReactWrapper {}`;

exports[`App snapshot renders 1`] = `ReactWrapper {}`;
